### PR TITLE
Upgrade to Checker Framework 3.4.0

### PR DESCRIPTION
**Pull request by nimakarimipour**
This version, uses the latest version of Checker Framework(3.4.0), and removes the local "Retunrs Receiver Checker" and uses the integrated checker inside Checker Framework.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
+#Wed May 06 13:44:48 CDT 2020
```
Comment (msridhar) [inline gradle/wrapper/gradle-wrapper.properties:1]: This is an unrelated change.  Please leave this file unchanged in this PR

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -162,23 +161,22 @@ private ReturnsRcvrAnnotatedTypeFactory getReturnsRcvrAnnotatedTypeFactory() {
...
 
-  /**
-   * Continue to trust but not check the old {@link
-   * org.checkerframework.checker.builder.qual.ReturnsReceiver} annotation, for
-   * backwards-compatibility.
-   */
-  private boolean hasOldReturnsReceiverAnnotation(MethodInvocationTree tree) {
-    return this.getDeclAnnotation(TreeUtils.elementFromUse(tree), ReturnsReceiver.class) != null;
-  }
+//  /**
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java]: Remove, don't comment out

Comment (nimakarimipour) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java]: Actually it was my mistake commenting this part, this method is no longer commented.

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java [hunk 3]
```diff
@@ -171,11 +170,7 @@ boolean returnsThis(final MethodInvocationTree tree) {
         || hasOldReturnsReceiverAnnotation(tree);
   }
 
-  /**
-   * Continue to trust but not check the old {@link
-   * org.checkerframework.checker.builder.qual.ReturnsReceiver} annotation, for
-   * backwards-compatibility.
-   */
+  /** Continue to trust but not check the old */
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java:173]: Looks like we lost some of the old documentation comment?  Can we restore?

Comment (nimakarimipour) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java:173]: Restored it. Now it's linked to the new location of ReturnsReceiver annotation.

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java [hunk 4]
```diff
@@ -79,18 +85,10 @@
...
   @Override
-  public void report(final Result r, final Object src) {
-
-    Result theResult = r;
-
-    String errKey = r.getMessageKeys().iterator().next();
-    if ("method.invocation.invalid".equals(errKey) && r.isFailure()) {
-      Object[] args = r.getDiagMessages().iterator().next().getArgs();
+  public void reportError(Object source, @CompilerMessageKey String messageKey, Object... args) {
+    String errKey = messageKey;
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java:90]: @kelloggm was there some other way you wanted to implement the custom error message, based on some new Checker Framework method?

Comment (kelloggm) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java:90]: Yes, but I can make that change separately after this lands if that's easier for you all.

Comment (msridhar) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java:90]: That's fine, as long as we don't forget

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java [hunk 5]
```diff
@@ -173,7 +172,7 @@ boolean returnsThis(final MethodInvocationTree tree) {
 
   /**
    * Continue to trust but not check the old {@link
-   * org.checkerframework.checker.builder.qual.ReturnsReceiver} annotation, for
+   * org.checkerframework.checker.returnsreceiver.builder.qual.ReturnsReceiver} annotation, for
```
Comment (kelloggm) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java:175]: This annotation's package cannot change. It's generated by Lombok, so we have to continue to use the old fully-qualified name here. Please restore it; it can be part of object-construction-qual if necessary.

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/LombokSupport.java [hunk 6]
```diff
@@ -116,7 +116,9 @@ private void handleToBuilderType(
     List<String> requiredProperties = getLombokRequiredProperties(classElement);
     AnnotationMirror calledMethodsAnno =
         atypeFactory.createCalledMethods(requiredProperties.toArray(new String[0]));
-    type.replaceAnnotation(calledMethodsAnno);
+    if (!type.isAnnotatedInHierarchy(calledMethodsAnno)) {
+      type.addAnnotation(calledMethodsAnno);
+    }
```
Comment (kelloggm) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/LombokSupport.java:121]: What is the purpose of this change? It seems unrelated, and I would expect it to cause tests to fail.

Comment (nimakarimipour) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/LombokSupport.java]: This was related to the changes in the Returns Receiver Checker but now it seems unnecessary, and restored to the original version.

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java [hunk 7]
```diff
@@ -171,6 +170,7 @@ boolean returnsThis(final MethodInvocationTree tree) {
         || hasOldReturnsReceiverAnnotation(tree);
   }
 
+  // org.checkerframework.checker.builder.qual.ReturnsReceiver
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java:173]: @nimakarimipour delete this?

File: object-construction-checker/tests/autovalue/Inheritance.java [hunk 8]
```diff
@@ -1,7 +1,6 @@
 import com.google.auto.value.AutoValue;
 import org.checkerframework.checker.objectconstruction.qual.*;
-import org.checkerframework.checker.returnsrcvr.qual.This;
-
```
Comment (kelloggm) [inline object-construction-checker/tests/autovalue/Inheritance.java:4]: Why delete this blank line?

File: object-construction-qual/src/main/java/org/checkerframework/checker/builder/qual/ReturnsReceiver.java [hunk 9]
```diff
@@ -8,15 +8,7 @@
...
- * receiver object.
- *
- * <p>This annotation can only be written on a method declaration. It is inherited by all overrides
- * of that method.
- *
- * <p>This annotation has been replaced by {@link
- * org.checkerframework.checker.returnsrcvr.qual.This}. It is retained only for
- * backwards-compatibility, including with Lombok's checkerframework = true lombok.config flag. It
- * should not be used in new code, because it is TRUSTED, NOT CHECKED.
+ * receiver object. should not be used in new code, because it is TRUSTED, NOT CHECKED.
```
Comment (kelloggm) [inline object-construction-qual/src/main/java/org/checkerframework/checker/builder/qual/ReturnsReceiver.java:11]: Why was this deleted?

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java [hunk 10]
```diff
@@ -79,18 +85,10 @@
     return result;
   }
 
-  /**
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionChecker.java:82]: We shouldn't delete this Javadoc comment
